src/downshift.js

```diff
@@ -502,7 +506,7 @@ class Downshift extends Component {
     },
 
     Enter(event) {
-      if (this.getState().isOpen) {
+      if (this.isOpen) {
```

Comment:
Was this an intentional change? `isOpen` is not an instance property, it is pull from the internal state, right?

Reply:
By reverting it to how it originally was, `getState` needs to be exposed from the object returned by `getStateAndHelpers()`. This is necessary because `getStateAndHelpers` object is always bound:

```
  input_handleKeyDown = event => {
    const handler =
      this.props.keyDownHandlers[event.key] || this.keyDownHandlers[event.key]
    if (handler) {
      handler.call(this.getStateAndHelpers(), event) // always bound
    }
  }
```

I am running into a wall after this though. `getState()` throws an error because at some point, `this.state` is `undefined`:

<img width="435" alt="screen shot 2018-02-26 at 11 34 13 am" src="https://user-images.githubusercontent.com/1811365/36691061-03800ba6-1ae9-11e8-837b-0cfc9c3a3bc4.png">

I will keep investigating this issue in the mean time ⌛️ .

Reply:
I totally forgot that we were doing `.call(this.getStateAndHelpers(), event)` and thought we were doing `.call(this, event)`. So your changes make total sense.

I'm thinking that I'd like to change the implementation slightly though. Could we do this instead?

```javascript
handler(event, this.getStateAndHelpers())
```

It doesn't resemble a true event handler anymore (because those only receive a single argument), but I think it makes more sense than using `this`. People could give things a much better name:

```javascript
const keyDownHandlers = {
  Enter(event, downshift) {
    if (downshift.isOpen) {
      // do something
    }
  }
}
```

What do you think?